debezium-server-voyager/debezium-server-voyagerexporter/src/main/java/io/debezium/server/ybexporter/EventQueue.java

```diff
@@ -111,56 +113,43 @@ private void recoverStateFromDisk() {
-     * queue segment, which would again write an EOF marker, leading to two EOF
-     * markers sequentially.
-     * Voyager import data will proceed to next segment upon seeing the first EOF so
-     * it's not a concern.
+     * This function gets the latest queue segment that was written to from
+     * queue_segment_meta table
+     * If no queue segments are found, it logs a message and returns.
+     * If the current queue segment has alreaady been closed by ./ we rotate to the
+     * next one
      */
     private void recoverLatestQueueSegment() {
-        // read dir to find all queue files
-        Path queueDirPath = Path.of(dataDir, QUEUE_FILE_DIR);
-        String searchGlob = String.format("%s.[0-9]*.%s", QUEUE_SEGMENT_FILE_NAME, QUEUE_SEGMENT_FILE_EXTENSION);
-        ArrayList<Path> filePaths = new ArrayList<>();
-        try {
-            DirectoryStream<Path> stream = Files.newDirectoryStream(queueDirPath, searchGlob);
-            for (Path entry : stream) {
-                filePaths.add(entry);
-            }
-            if (filePaths.size() == 0) {
-                // no files found. nothing to recover.
-                LOGGER.info("No files found matching {}. Nothing to recover", searchGlob);
-                return;
-            }
-            // extract max index of all files
-            long maxIndex = 0;
-            Path maxIndexPath = null;
-            for (Path p : filePaths) {
-                String filename = p.getFileName().toString();
-                String indexStr = filename.substring(QUEUE_SEGMENT_FILE_NAME.length() + 1,
-                        filename.length() - (QUEUE_SEGMENT_FILE_EXTENSION.length() + 1));
-                long index = Long.parseLong(indexStr);
-                if (index >= maxIndex) {
-                    maxIndex = index;
-                    maxIndexPath = p;
-                }
+        long fetchedQueueSegmentIndex = es.getLastQueueSegmentIndex();
+        String fetchedQueueSegmentExporterRole = es.getLastQueueSegmentExporterRole();
+        if (fetchedQueueSegmentIndex == -1) {
+            LOGGER.info("No queue segments found. Nothing to recover");
+            return;
+        }
+        currentQueueSegmentIndex = fetchedQueueSegmentIndex;
+
+        if (fetchedQueueSegmentExporterRole.equals("source_db_exporter") && sourceType.equals("yb")) {
+            // if the latest queue segment was written by source_db_exporter and we are
+            // currently exporting from target, we need to
+            // rotate to the next one
+            currentQueueSegmentIndex++;
+            currentQueueSegment = new QueueSegment(dataDir, currentQueueSegmentIndex,
+                    getFilePathWithIndex(currentQueueSegmentIndex));
+        } else {
+            currentQueueSegment = new QueueSegment(dataDir, currentQueueSegmentIndex,
+                    getFilePathWithIndex(currentQueueSegmentIndex));
+            if (currentQueueSegment.isClosed()) {
```

Comment:
Again, this should already be handled in `recoverStateFromDisk` . Pls test once

Reply:
Removed this